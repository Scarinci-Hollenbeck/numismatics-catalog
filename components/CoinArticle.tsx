import { create } from 'domain';
import React from 'react';
import { Article } from '../interfaces';
import ButtonLink from './ButtonLink';
import { createMarkup, formatDate } from '../utils/helpers';


type Props = {
  article: Article
}

export default function CoinArticle({ article }: Props): JSX.Element {

  return (
    <li>      
      <div>
        <h5>{article.node.title}</h5>
        <p><strong>Published: </strong>{formatDate(article.node.date)}</p>
        <div
          className="excerpt"
          dangerouslySetInnerHTML={createMarkup(article.node.excerpt)}
        />
        <ButtonLink
          link={`https://donaldscarinci.com${article.node.uri}`}
          caption="Full Article"
        />
      </div>
      <style jsx>
        {`
          li {
            font-family: 'Tajawal Regular';
            list-style-type: none;
            margin: 0;
            margin-bottom: 2em;
            padding: 0;
            text-align: left;
            padding-bottom: 2em;
            color: #fff;            
          }  

          .excerpt {
            margin-left: 0;
            margin-bottom: 2em;
          }

          h5 {
            font-size: 1.7rem;
          }

          p,
          h5 {
            margin: 0;
            padding: 0;
          }

          a {
            font-weight: bold;
            font-size: 1.2rem;
          }

          @media (min-width: 1225px) {
            li {
              margin-right: 3em;
            }
          }
        `}
      </style>
    </li>
  );
}
