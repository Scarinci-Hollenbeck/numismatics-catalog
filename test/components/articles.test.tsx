import React from 'react'
import renderer from 'react-test-renderer'
import CoinArticle from '../../components/CoinArticle'
import CoinArticleContainer from '../../components/CoinArticlesCointainer'

const articles = [
  {
    id: '123',
    title: 'Austria’s Named Best Gold Coin at 2020 COTY Awards',
    description:
      'The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in',
    image:
      'https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png',
    link:
      'https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/',
  },
  {
    id: '456',
    title: 'Austria’s Named Best Gold Coin at 2020 COTY Awards',
    description:
      'The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in',
    image:
      'https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png',
    link:
      'https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/',
  },
  {
    id: '789',
    title: 'Austria’s Named Best Gold Coin at 2020 COTY Awards',
    description:
      'The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in',
    image:
      'https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png',
    link:
      'https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/',
  },
  {
    id: '912',
    title: 'Austria’s Named Best Gold Coin at 2020 COTY Awards',
    description:
      'The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in',
    image:
      'https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png',
    link:
      'https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/',
  },
  {
    id: '913',
    title: 'Austria’s Named Best Gold Coin at 2020 COTY Awards',
    description:
      'The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in',
    image:
      'https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png',
    link:
      'https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/',
  },
]

describe('Articles Item component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<CoinArticle article={articles[0]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Articles Container component', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(<CoinArticleContainer articles={articles} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
