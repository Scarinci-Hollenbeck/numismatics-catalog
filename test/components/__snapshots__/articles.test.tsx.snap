// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Container component matches snapshot 1`] = `
<section
  className="coin-articles"
>
  <h2>
    More On Numismatics
  </h2>
  <hr />
  <ul>
    <li
      className="article-item"
    >
      <img
        alt="Austria’s Named Best Gold Coin at 2020 COTY Awards"
        src="https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png"
      />
      <div
        className="article-content"
      >
        <h5>
          Austria’s Named Best Gold Coin at 2020 COTY Awards
        </h5>
        <p>
          The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in
        </p>
        <a
          href="https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/"
          rel="noopener"
          target="_blank"
        >
          Read Full Article &gt;&gt;
        </a>
      </div>
    </li>
    <li
      className="article-item"
    >
      <img
        alt="Austria’s Named Best Gold Coin at 2020 COTY Awards"
        src="https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png"
      />
      <div
        className="article-content"
      >
        <h5>
          Austria’s Named Best Gold Coin at 2020 COTY Awards
        </h5>
        <p>
          The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in
        </p>
        <a
          href="https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/"
          rel="noopener"
          target="_blank"
        >
          Read Full Article &gt;&gt;
        </a>
      </div>
    </li>
    <li
      className="article-item"
    >
      <img
        alt="Austria’s Named Best Gold Coin at 2020 COTY Awards"
        src="https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png"
      />
      <div
        className="article-content"
      >
        <h5>
          Austria’s Named Best Gold Coin at 2020 COTY Awards
        </h5>
        <p>
          The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in
        </p>
        <a
          href="https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/"
          rel="noopener"
          target="_blank"
        >
          Read Full Article &gt;&gt;
        </a>
      </div>
    </li>
    <li
      className="article-item"
    >
      <img
        alt="Austria’s Named Best Gold Coin at 2020 COTY Awards"
        src="https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png"
      />
      <div
        className="article-content"
      >
        <h5>
          Austria’s Named Best Gold Coin at 2020 COTY Awards
        </h5>
        <p>
          The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in
        </p>
        <a
          href="https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/"
          rel="noopener"
          target="_blank"
        >
          Read Full Article &gt;&gt;
        </a>
      </div>
    </li>
    <li
      className="article-item"
    >
      <img
        alt="Austria’s Named Best Gold Coin at 2020 COTY Awards"
        src="https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png"
      />
      <div
        className="article-content"
      >
        <h5>
          Austria’s Named Best Gold Coin at 2020 COTY Awards
        </h5>
        <p>
          The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in
        </p>
        <a
          href="https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/"
          rel="noopener"
          target="_blank"
        >
          Read Full Article &gt;&gt;
        </a>
      </div>
    </li>
  </ul>
</section>
`;

exports[`Articles Item component matches snapshot 1`] = `
<li
  className="article-item"
>
  <img
    alt="Austria’s Named Best Gold Coin at 2020 COTY Awards"
    src="https://donaldscarinci.com/wp-content/uploads/2020/01/Austria-Alfred-Alder-50-Euro-Gold.png"
  />
  <div
    className="article-content"
  >
    <h5>
      Austria’s Named Best Gold Coin at 2020 COTY Awards
    </h5>
    <p>
      The Austrian Mint won the “Best Gold Coin” category at 2020 Coin of the Year (COTY) Awards for its 50 Euro honoring Alfred Adler. The winning coin, dated 2018, is part of a three-coin series that celebrates the pioneering schools of psychotherapy. The other coins in
    </p>
    <a
      href="https://donaldscarinci.com/austrias-named-best-gold-coin-at-2020-coty-awards/"
      rel="noopener"
      target="_blank"
    >
      Read Full Article &gt;&gt;
    </a>
  </div>
</li>
`;
